### PR TITLE
feat: support search NFT and tokens【6.4】

### PR DESCRIPTION
**Pull request by dudu0506**
https://mask.atlassian.net/browse/FW-2703

# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/app/(normal)/intent/compose/pages/ShareLinkPage.tsx [hunk 1]
```diff
@@ -56,9 +56,9 @@ const fireflyMention = {
 
 async function searchIdentities(query: string) {
     const data = await FireflyEndpointProvider.searchIdentity(query);
```
Comment (guanbinrui) [inline src/app/(normal)/intent/compose/pages/ShareLinkPage.tsx:58]: resolveFireflyResponseData

File: src/app/(normal)/search/[source]/[type]/layout.tsx [hunk 2]
```diff
@@ -11,5 +12,8 @@ export async function generateMetadata() {
 }
 
 export default function SearchLayout({ children }: PropsWithChildren) {
-    return <div>{children}</div>;
+    return <div>
```
Comment (guanbinrui) [inline src/app/(normal)/search/[source]/[type]/layout.tsx:15]: prettier code

File: src/components/Lexical/plugins/AtMentionsPlugin.tsx [hunk 3]
```diff
@@ -188,10 +188,10 @@ export function MentionsPlugin(): JSX.Element | null {
         ],
         queryFn: async () => {
             if (!debounceQuery) return;
-            const data = await FireflyEndpointProvider.searchIdentity(debounceQuery, availableSources);
+            const data = await FireflyEndpointProvider.searchIdentity(debounceQuery);
```
Comment (guanbinrui) [inline src/components/Lexical/plugins/AtMentionsPlugin.tsx:191]: resolveFireflyResponseData

File: src/components/Search/SearchSources.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            const supportTypes = SORTED_SEARCH_TYPE[narrowToSocialSource(x)];
+            if (!supportTypes.includes(searchType)) return false;
+
+            return true;
+        });
+    }, [isTwitterLoggedIn, searchType]);
+
+    return (
+        <nav className="mt-3 flex gap-x-2 px-4">
+            {sources.map((source) => {
```
Comment (guanbinrui) [inline src/components/Search/SearchSources.tsx:40]: SourceTab

File: src/components/Lexical/plugins/AtMentionsPlugin.tsx (hunk 3 above)
Comment (dudu0506) [reply src/components/Lexical/plugins/AtMentionsPlugin.tsx:191]: not suitable here

File: src/app/(normal)/intent/compose/pages/ShareLinkPage.tsx (hunk 1 above)
Comment (dudu0506) [reply src/app/(normal)/intent/compose/pages/ShareLinkPage.tsx:58]: not suitable here, its Pageable

**Review by guanbinrui — APPROVED** [review]

File: src/app/(normal)/search/[...slug]/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    return false;
+}
+
+export default function SearchLayout({
+    params,
+    children,
+}: PropsWithChildren<{
+    params: { slug: string[] };
+}>) {
+    if (!checkSlug(params.slug)) return notFound();
```
Comment (guanbinrui) [inline src/app/(normal)/search/[...slug]/layout.tsx:37]:
```suggestion
    if (!checkSlug(params.slug)) notFound();
```

File: src/components/Search/SearchRecommendation.tsx [hunk 6]
```diff
@@ -82,15 +98,15 @@ export function SearchRecommendation(props: SearchRecommendationProps) {
...
+    if (keyword && !isSearchPage) {
         return (
             <div className={containerClasses}>
                 <h2 className="p-3 pb-2 text-sm">
                     <Trans>Publications</Trans>
                 </h2>
                 <Link
                     className="flex cursor-pointer items-center px-4 py-4 text-left hover:bg-bg"
-                    href={resolveSearchUrl(keyword)}
+                    href={fixSearchUrl(keyword)}
```
Comment (guanbinrui) [inline src/components/Search/SearchRecommendation.tsx:109]:
```suggestion
                    href={fixSearchUrl(isSearchPage, query, searchType, source)}
```

File: src/app/(normal)/search/[...slug]/page.tsx [hunk 7]
```diff
@@ -87,7 +118,14 @@ export default function Page() {
             return lastPage?.nextIndicator?.id;
         },
         select(data) {
-            return compact(data.pages.flatMap((x) => (x?.data ?? []) as Array<Profile | Post | Channel>));
+            return compact(
+                data.pages.flatMap(
+                    (x) =>
+                        (x?.data ?? []) as Array<
```
Comment (guanbinrui) [inline src/app/(normal)/search/[...slug]/page.tsx:124]: Please fix type of `searchTokens`.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**